modin/experimental/sql/hdk/query.py

```diff
@@ -66,7 +67,14 @@ def hdk_query(query: str, **kwargs) -> pd.DataFrame:
     worker = HdkWorker()
     if len(kwargs) > 0:
         query = _build_query(query, kwargs, worker.import_arrow_table)
-    return from_arrow(worker.executeDML(query))
+    at = worker.executeDML(query)
+    schema = at.schema
+    # HDK returns strings as dictionary. Cast dictionary columns to string.
+    if cast := {i: f for i, f in enumerate(schema) if pa.types.is_dictionary(f.type)}:
```

Comment:
so we unconditionally cast all dictionary columns to strings, what if there were columns that have to be a dictionary encoded?
```python
mdf = mdf.astype("category")
res = execute_some_query(mdf)
res.dtypes == "string" # ?
```

could please you also measure what it cost us to cast the columns to strings? it feels to be very expensive. Can we implement something like a delayed casting? 

As far as I understand there's nothing wrong for strings to be dictionary encoded unless we're casting the frame to pandas. So can we just flag the columns we want to cast to strings in the future and do so only when conversion to pandas happens?

Reply:
Initially, I've implemented the cast only in to_pandas(), but the SQL tests failed. The problem here is that we don't know the real types.

Reply:
I measured the cast time on my laptop, running WSL2, using the below code:
```python
from modin.experimental.sql import query
data = {str(k): [str(v) for v in range(10000)] for k in range(100)}
df = pd.DataFrame(data)
print(query("SELECT * FROM df", df=df))
```
Time: 0.01735997200012207

Reply:
Replaced cast with fake dtypes.